mindmeld/components/question_answerer.py

```diff
@@ -227,6 +236,23 @@ def load_kb(self, index_name, data_file, **kwargs):
             raise ValueError(msg)
         self._load_kb(index_name, data_file, **kwargs)
 
+    def dump(self, pkl_path):
+        folder = os.path.dirname(pkl_path)
+        os.makedirs(folder, exist_ok=True)
+        self._dump(pkl_path)
+
+        metadata = {
+            "app_path": self.app_path,
```

Comment:
One issue with this implementation is `_dump ` for `NativeQuestionAnswerer` reference `ALL_INDICES`, which is a class variable. So if multiple apps are loaded into memory at the same time, `ALL_INDICES` will be updated with all of them, not just the one related to this app_path. So the metadata will not accurately partition the indices based on app path.